src/effects/onFailure/test.js

```diff
@@ -0,0 +1,45 @@
+import Immutable from 'seamless-immutable';
+
+import createAction from '../../creators/createAction';
+import createReducer from '../../creators/createReducer';
+
+import onFailure from '.';
+
+const initialState = {
+  target: 'Some content',
+  targetLoading: true,
+  targetError: null
+};
+
+const setUp = {
+  state: null
+};
+
+beforeEach(() => {
+  setUp.state = Immutable(initialState);
+});
+
+describe('onFailure', () => {
+  it('Sets correctly error and loading', () => {
+    const reducer = createReducer(setUp.state, {
+      '@@ACTION/TYPE': onFailure()
+    });
+    const newState = reducer(setUp.state, createAction('@@ACTION/TYPE', 'target', 'Oops !'));
+    expect(newState).toEqual({
+      target: 'Some content',
+      targetLoading: false,
+      targetError: 'Oops !'
+    });
+  });
+  it('Sets conditionally error content', () => {
+    const reducer = createReducer(setUp.state, {
+      '@@ACTION/TYPE': onFailure(action => `Error: ${action.payload}`)
+    });
+    const newState = reducer(setUp.state, createAction('@@ACTION/TYPE', 'target', 'Oops !'));
```

Comment:
This is something I've thought in several opportunities while reviewing this pull request.
By using different params for the action type, action target and action payload you force the developer to know the interface of method (i mean the order of the params and what they do) which makes the learning curve higher.
Maybe, if createAction receives an object { type, payload, target}, the library will be easier to use because each time you read the code, you will know `createAction` do, because the params are named.
When I read this line, I had to go back to the `createAction` definition to remeber what does it do